pkg/machines/components/notification/notificationArea.jsx

```diff
@@ -0,0 +1,58 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { PropTypes } from "react";
+import { Notification, NotificationMessage } from "./notification.jsx";
+
+React;
+
+const NotificationArea = ({ notifications, onDismiss, id }) => {
+    if (!notifications || notifications.length === 0) {
+        return null;
+    }
+
+    const notificationList = [...notifications].sort((a, b) => a.id < b.id).map((notification) => {
```

Comment:
It feels a bit odd to sort by this internal and invisible ID instead of keeping the order they were reported. The IDs are generated as ascending numbers, so this sorting should never actually change anything (even with discarded ones).

Reply:
yes they are in ascending order, but I do this to have them in descending order so the newest notifications appear at the top